includes/forum-reactions.php

```diff
@@ -169,36 +174,43 @@ public function reaction_change($post_id, $user_id, $reaction) {
 
         // Add reaction when there is none.
         if ($reaction_check === false) {
-            $this->add_reaction($post_id, $user_id, $reaction);
+            $this->add_reaction($post_id, $user_id, $reaction, $author);
         }
 
         // Remove reaction when it is already set.
         if ($reaction_check === $reaction) {
-            $this->remove_reaction($post_id, $user_id, $reaction);
+            $this->remove_reaction($post_id, $user_id, $reaction, $author);
         }
 
         // Update reaction when it is different.
         if ($reaction_check !== $reaction) {
-            $this->update_reaction($post_id, $user_id, $reaction);
+            $this->update_reaction($post_id, $user_id, $reaction, $author);
         }
 
         return true;
     }
 
-    public function add_reaction($post_id, $user_id, $reaction) {
-        $this->asgarosforum->db->insert($this->asgarosforum->tables->reactions, array('post_id' => $post_id, 'user_id' => $user_id, 'reaction' => $reaction), array('%d', '%d', '%s'));
+    public function add_reaction($post_id, $user_id, $reaction, $author) {
+		// Get the current time.
+        $date = $this->asgarosforum->current_time();
+
+        $this->asgarosforum->db->insert($this->asgarosforum->tables->reactions, array('post_id' => $post_id, 'user_id' => $user_id, 'reaction' => $reaction, 'author' => $author, 'datestamp' => $date), array('%d', '%d', '%s', '%d', '%s'));
 
         do_action('asgarosforum_after_add_reaction', $post_id, $user_id, $reaction);
     }
 
-    public function remove_reaction($post_id, $user_id, $reaction) {
-        $this->asgarosforum->db->delete($this->asgarosforum->tables->reactions, array('post_id' => $post_id, 'user_id' => $user_id, 'reaction' => $reaction), array('%d', '%d', '%s'));
+    public function remove_reaction($post_id, $user_id, $reaction, $author) {
+
+        $this->asgarosforum->db->delete($this->asgarosforum->tables->reactions, array('post_id' => $post_id, 'user_id' => $user_id, 'reaction' => $reaction, 'author' => $author), array('%d', '%d', '%s', '%d'));
```

Comment:
I dont think we need the `$author` variable to remove the reaction because the combination of `$post_id` and `$user_id` already ensures that it is a unique combination.

Reply:
You are absolutely right.
I will revert this chaneg.